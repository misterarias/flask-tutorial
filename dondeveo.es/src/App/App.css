body {
  font-family: Roboto, helvetica, arial, sans-serif;
}

.border {
  height: 100vh;
  border: 0 ;
}

.jumbotron {
  background: #fdffff;
  padding-bottom: 1rem;
}
