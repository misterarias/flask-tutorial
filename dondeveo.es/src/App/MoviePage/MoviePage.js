import React from 'react';
import { Row } from 'reactstrap';
import './MoviePage.css' ;

const MoviePage = (props) => {
  return (
    <Row>
      <h1>Soy una muvi</h1>
    </Row>
  );
}

export default MoviePage;
