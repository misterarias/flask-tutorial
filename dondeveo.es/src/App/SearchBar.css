.container {
  text-align: center;
  background-color: white;
  padding: 10px;
  margin: 60px;
}

.header {
  height: 60px;
  padding: 10px;
  color: #222;
}

.title {
  font-size: 1.5em;
}

.searchbar input {
  width: 150px;
  height: 40px;
  padding-left: 5px;
  margin: 0 5px;
  font-size: 16px;
}
.searchbar button {
  height: 40px;
  font-size: 16px;
}
