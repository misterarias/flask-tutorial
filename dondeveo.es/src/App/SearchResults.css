.search-results {
  background: #cebada;
}

.img-circle {
  /* make it responsive */
  max-width: 100%;
  width: 100%;
  /* height: 100%; */
  display: block;

  /* make it a circle */
  padding: 50%;
  background: #e1e1e1;
  border-radius: 50%;

  /* Centering on image`s center*/
  background-position-y: center;
  background-position-x: center;
  background-repeat: no-repeat;

  /* it makes the clue thing, takes smaller dimension to fill div */
  background-size: cover;

  /* it is optional, for making this div centered in parent*/
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}
