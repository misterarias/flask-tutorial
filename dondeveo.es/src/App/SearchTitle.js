import React from 'react';
import { Row, Jumbotron } from 'reactstrap';
import './SearchTitle.css' ;

const SearchTitle = (props) => {
  return (
    <Row>
      <Jumbotron>
        <h1 className="display-3">dondeveo.es</h1>

        <p class="lead opacity-75">
          Busca aqu√≠ donde puedes ver la peli que te apetezca
        </p>

      </Jumbotron>
    </Row>
  );
}

export default SearchTitle;
