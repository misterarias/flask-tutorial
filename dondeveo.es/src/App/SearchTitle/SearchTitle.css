.opacity-75 {
  opacity:0.75;
}

.jumbotron {
  background: #fdffff;
  padding-bottom: 1rem;
}
