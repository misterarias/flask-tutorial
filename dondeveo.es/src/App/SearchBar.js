import React, { Component } from 'react';
import './SearchBar.css';

export default class SearchBar extends Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

    this.state = {
      query: ""
    }
  }

  handleChange(ev) {
    this.setState({
      query: ev.target.value
    });
  }

  handleSubmit(ev) {
    this.props.onSubmit(this.state.query);
  }

  render() {
    return (
      <div className="container">

        <header className="header">
          <h1 className="title">¿Dónde veo...?</h1>
        </header>

        <section className="searchbar">

          <input
            type="text"
            onChange={this.handleChange}
            context="Search"
          />

          <button
            className="submit"
            onClick={this.handleSubmit}
          >GO</button>

        </section>
      </div>
    )
  }
}
