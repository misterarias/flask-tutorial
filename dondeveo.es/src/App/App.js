import React, { Component } from 'react';
import './App.css';

import SearchBar from './SearchBar' ;
import SearchResults from './SearchResults' ;

export default class App extends Component {
  render() {
    return (
      <div className="App">
          <SearchBar />
          <SearchResults />
        </div>
    );
  }
}
