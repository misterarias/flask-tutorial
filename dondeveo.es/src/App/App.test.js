import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

import MoviePage from './MoviePage/MoviePage' ;
import SearchBar from './SearchBar/SearchBar' ;
import SearchTitle from './SearchTitle/SearchTitle' ;

import { MemoryRouter} from 'react-router-dom'

import { mount, configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

it('renders default path without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter >
      <App />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders "/movies"  path without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/movie/Rambo' ]}  >
      <App />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('App has the expected elements when on the default path', () => {
  const app = mount(
    <MemoryRouter >
      <App />
    </MemoryRouter>
  );

  expect(app.find(SearchTitle).length).toBe(1);
  expect(app.find(SearchBar).length).toBe(1);
  expect(app.find(MoviePage).length).toBe(0);
});

test('App has the expected elements when on the "/movies" path', () => {
  const app = mount(
    <MemoryRouter initialEntries={[ '/movie/Rambo' ]}>
      <App />
    </MemoryRouter>
  );

  expect(app.find(SearchTitle).length).toBe(1);
  expect(app.find(SearchBar).length).toBe(0);
  expect(app.find(MoviePage).length).toBe(1);
});
