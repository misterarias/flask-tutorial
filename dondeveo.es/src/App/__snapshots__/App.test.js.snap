// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <header
      className="header"
    >
      <h1
        className="title"
      >
        ¿Dónde veo...?
      </h1>
    </header>
    <section
      className="searchbar"
    >
      <input
        context="Search"
        onChange={[Function]}
        type="text"
      />
      <button
        className="submit"
        onClick={[Function]}
      >
        GO
      </button>
    </section>
  </div>
  <div
    className="search-results"
  >
    <ul
      id="results"
    />
  </div>
</div>
`;
